internal/infrastructure/kubernetes/proxy/resource.go

```diff
@@ -77,12 +77,14 @@ func EnvoyAppLabelSelector() []string {
 }
 
 // envoyLabels returns the labels, including extraLabels, used for Envoy resources.
-func envoyLabels(extraLabels map[string]string) map[string]string {
+func envoyLabels(extraLabels, customLabels map[string]string) map[string]string {
```

Comment:
this is not a great API imo
the caller should be merging this maybe using https://pkg.go.dev/golang.org/x/exp/maps#Copy before passing it to `envoyLabels`

Reply:
fixed